File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/CategoricalAxis.swift
//
//  CategoricalAxis.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class CategoricalAxis:TKExamplesExampleViewController {

    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        // >> chart-category-axis-swift
        var array = [TKChartBubbleDataPoint]()
        let categories = ["Apple", "Google", "Microsoft", "Samsung"]
        
        for item in 0..<categories.count {
            array.append(TKChartBubbleDataPoint(x: categories[item], y: Int(arc4random()%100), area: 0))
        }
        
        let series = TKChartColumnSeries(items:array)
        series.selection = TKChartSeriesSelection.Series
        
        // >> chart-add-axis-swift
        let xAxis = TKChartCategoryAxis(categories:categories)
        xAxis.position = TKChartAxisPosition.Bottom
        xAxis.setPlotMode(TKChartAxisPlotMode.BetweenTicks)
        xAxis.range = TKRange(minimumIndex: 0, andMaximumIndex: 3)
        chart.xAxis = xAxis
        // << chart-add-axis-swift
        // << chart-category-axis-swift
        
        let yAxis = TKChartNumericAxis(minimum:0, andMaximum:100)
        yAxis.position = TKChartAxisPosition.Left
        chart.yAxis = yAxis
        
        chart.addSeries(series)
        
        // >> chart-title-swift
        chart.xAxis!.title = "Vendors"
        chart.xAxis!.style.titleStyle.textColor = UIColor.grayColor()
        chart.xAxis!.style.titleStyle.font = UIFont.boldSystemFontOfSize(11)
        chart.xAxis!.style.titleStyle.alignment = TKChartAxisTitleAlignment.RightOrBottom
        chart.reloadData()
        // << chart-title-swift
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

File: Examples/Examples/Docs/AxisTypes/CategoricalAxis.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("CategoricalAxis")]
	public class CategoricalAxis: XamarinExampleViewController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			TKChart chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			Random r = new Random ();

			// >> chart-category-axis-cs
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			string[] categories = new []{"Apple", "Google", "Microsoft", "Samsung"};
			for (int i = 0; i < categories.Length; i++) {
				list.Add(new TKChartDataPoint(new NSString(categories[i]), new NSNumber(r.Next() % 100))); 
			}

			TKChartColumnSeries series = new TKChartColumnSeries (list.ToArray());
			series.SelectionMode = TKChartSeriesSelectionMode.Series;

			// >> chart-add-axis-cs
			TKChartCategoryAxis xAxis = new TKChartCategoryAxis ();
			xAxis.Position = TKChartAxisPosition.Bottom;
			xAxis.PlotMode = TKChartAxisPlotMode.BetweenTicks;
			series.XAxis = xAxis;
			// << chart-add-axis-cs
			// << chart-category-axis-cs

			TKChartNumericAxis yAxis = new TKChartNumericAxis (new NSNumber (0), new NSNumber (100));
			yAxis.Position = TKChartAxisPosition.Left;
			chart.YAxis = yAxis;

			chart.AddSeries (series);

			// >> chart-title-cs
			xAxis.Title = "Vendors";
			xAxis.Style.TitleStyle.TextColor = UIColor.Gray;
			xAxis.Style.TitleStyle.Font = UIFont.BoldSystemFontOfSize (11);
			xAxis.Style.TitleStyle.Alignment = TKChartAxisTitleAlignment.RightOrBottom;
			chart.ReloadData ();
			// << chart-title-cs
		}
	}
}

